File: javascript/easy/maxConsecutiveOnes.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let maxConsecutiveOnes = 0, count = 0;
    for(let num of nums) {
        if(num === 1) {
            count++;
        } else {
            maxConsecutiveOnes = Math.max(maxConsecutiveOnes, count);
            count = 0;
        }
    }
    return Math.max(maxConsecutiveOnes, count);
};

File: ruby/easy/max_consecutive_ones.rb
# @param {Integer[]} nums
# @return {Integer}
def find_max_consecutive_ones(nums)
    count, max_consecutive_ones = 0, 0
    for i in 0..nums.length-1
        if nums[i] == 1
            count += 1
        else
            max_consecutive_ones = [max_consecutive_ones, count].max
            count = 0
        end
    end
    return [max_consecutive_ones, count].max
end
